.evergreen/utils.sh

```diff
@@ -43,6 +43,7 @@ find_python3() {
 createvirtualenv () {
     PYTHON=$1
     VENVPATH=$2
+    rm -rf $VENVPATH
```

Comment:
What was the motivation for this?

Reply:
[PYTHON-4570](https://jira.mongodb.org/browse/PYTHON-4570), I'll link it